File: src/Web/Models/Search/ProductSearchModel.cs
using Microsoft.AspNetCore.Mvc;
using System.Collections.Generic;

namespace Web.Models
{
    public class ProductSearch
    {
        [FromQuery(Name = "id")]
        public int[] Ids { get; set; }
        [FromQuery(Name = "title")]
        public string[] Titles { get; set; }
        [FromQuery(Name = "priceRange")]
        public PriceRangeModel PriceRange { get; set; }
        [FromQuery(Name = "groupId")]
        public int[] GroupIds { get; set; }
        [FromQuery(Name = "catalogId")]
        public int[] CatalogIds { get; set; }
        [FromQuery(Name = "brandId")]
        public int[] BrandIds { get; set; }
        [FromQuery(Name = "specification")]
        public KeyValuePair<int, string>[] Specifications { get; set; }
    }
}

File: src/Web/ClientApp/src/models/search/ProductSearchModel.ts
import {PriceRange} from '../PriceRange';

export class ProductSearchModel {
  constructor(
    public ids?: number[],
    public titles?: string[],
    public groupIds?: number[],
    public catalogIds?: number[],
    public brandIds?: number[],
    public priceRange?: PriceRange
  ) {
  }
}
